flink-table-store-common/src/main/java/org/apache/flink/table/store/file/predicate/LeafPredicate.java

```diff
@@ -84,7 +99,10 @@ public List<Object> literals() {
 
     @Override
     public boolean test(Object[] values) {
```

Comment:
Maybe we can provide an interface instead of an object array for the input, so we have the opportunity to create something similar to `CastedRow` to avoid affecting the `LeafPredicate` code.

WDYT?

Reply:
The added cast function is for predicate push down, but I found it won't be used when the `LeafPredicate` is converted to filter in file format such as orc. I have updated `SchemaEvolutionUtil.createDataFilters` to convert predicate value to underlying data value for different file format, please help to review it, HTX